### PR TITLE
Openmetadata workflow issue 9172

### PR DESCRIPTION
**Pull request by dhruvinmaniar123**
### Describe your changes :

I worked on the #issue9172 for creating a new workflow which build the openmetadata application and attach to the existing/new release based on the user inputs or by drafting the new release.

#
### Type of change :

- [x] New feature

#
### Caveat while running the workflow:

-  If script is being run manually by giving inputs make sure you select tag option instead of branch.
- Before giving inputs manually make sure that no existing app is attached in the release with same name otherwise the workflow will fail as it won't allow to overwrite the file. In this scenario please go to that release and delete the existing app and then only run the workflow.
- If you are drafting for new release version make sure to update those versions in workflow under the "release-project-event-release" job.

#

### CODE REVIEW

File: .github/workflows/docker-openmetadata-server.yml [hunk 1]
```diff
@@ -9,48 +9,146 @@
...
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      
+      - name: Setup Test Containers Properties
+        run: |
+          sudo make install_antlr_cli
+          echo 'testcontainers.reuse.enable=true' >> $HOME/.testcontainers.properties
```
Comment (akash-jain-10) [inline .github/workflows/docker-openmetadata-server.yml:44]: This is probably not required. Let's remove this one!

File: .github/workflows/docker-openmetadata-server.yml [hunk 2]
```diff
@@ -9,48 +9,146 @@
...
+
+      - name: Build PR with Maven
```
Comment (akash-jain-10) [inline .github/workflows/docker-openmetadata-server.yml:46]: Let's use proper step names here. `Build OpenMetadata Server Application`.

File: .github/workflows/docker-openmetadata-server.yml [hunk 3]
```diff
@@ -9,48 +9,146 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git fetch --no-tags https://github.com/open-metadata/OpenMetadata.git +refs/heads/main:refs/remotes/origin/main
```
Comment (akash-jain-10) [inline .github/workflows/docker-openmetadata-server.yml:50]: What is the reason for fetching the main branch here ? If this is not related, let's remove this one.

File: .github/workflows/docker-openmetadata-server.yml [hunk 4]
```diff
@@ -9,48 +9,146 @@
...
+          mvn versions:set -DnewVersion=0.13.0
```
Comment (akash-jain-10) [inline .github/workflows/docker-openmetadata-server.yml:51]: We shouldn't be doing this here. Let's remove. The assumption here will be the repo source will already have the version number updated!

File: .github/workflows/docker-openmetadata-server.yml [hunk 5]
```diff
@@ -9,48 +9,146 @@
...
+          mvn -DskipTests clean install
```
Comment (akash-jain-10) [inline .github/workflows/docker-openmetadata-server.yml:52]: Let's use `mvn -DskipTests clean package` instead

File: .github/workflows/docker-openmetadata-server.yml [hunk 6]
```diff
@@ -9,48 +9,146 @@
...
+          asset_name: openmetadata-${{ inputs.GITHUB_RELEASE_TAG }}.tar.gz
+          asset_content_type: application/tar.gz
+
   push_to_docker_hub:
     runs-on: ubuntu-latest
     env:
-      input: ${{ github.event.inputs.tag }}
-      
+      DOCKER_RELEASE_TAG: ${{ github.event.inputs.DOCKER_RELEASE_TAG }}
+    needs: [maven-build]
```
Comment (akash-jain-10) [inline .github/workflows/docker-openmetadata-server.yml:124]: This workflow should wait for file upload first to Github release and then start. Or else for release scenario, this will run in parallel with  `release-project-event-release` or `release-project-event-workflow_dispatch` job and will not get the updated binary required for building the docker image.

File: .github/workflows/docker-openmetadata-server.yml [hunk 7]
```diff
@@ -9,48 +9,146 @@
...
     types: [published]
-
 jobs:
+  maven-build:
+    runs-on: ubuntu-latest
+    steps:
+
+      # - name: Check trigger type
+      #   if: ${{ env.input == '' }}
+      #   run: echo "input=0.13.0" >> $GITHUB_ENV
```
Comment (akash-jain-10) [inline .github/workflows/docker-openmetadata-server.yml:31]: Let's remove the commented code here.

File: .github/workflows/docker-openmetadata-server.yml [hunk 8]
```diff
@@ -9,48 +9,146 @@
...
+      
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      
+      - name: Setup Test Containers Properties
```
Comment (akash-jain-10) [inline .github/workflows/docker-openmetadata-server.yml:41]: Let's update the step name to be `Install antlr cli`

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]
Thanks @dhruvinmaniar123 for the contribution. Left few comments for the code changes. Let's fix those and get this feature available.

**Comment by dhruvinmaniar123** [comment]
Thanks @akash-jain-10 for the feedback. Changes have been done and resolved.
